lib/charms/mongodb/v1/mongodb_vm_legacy_provider.py

```diff
@@ -64,6 +66,16 @@ def _on_legacy_relation_created(self, event):
             )
             return
 
+        if self.charm.is_role(Config.Role.SHARD) or self.charm.is_role(Config.Role.CONFIG_SERVER):
```

Comment:
This code block is similar to the one above, would probably be an argument in favor of externalizing rel related statuses in a separate method?

Reply:
This is an idea I am in favour of, we have many checks like this throughout the code. Let me think of a nice way to do this

Reply:
think I got it :)